packages/hash/subgraph/src/types/edge/outward-edge-alias.ts

```diff
@@ -0,0 +1,44 @@
+/**
+ * A collection of 'aliases' which describe various variants of outward edges in more accessible-forms
+ */
+import { KnowledgeGraphOutwardEdge } from "../edge";
+import { EntityIdAndTimestamp } from "../identifier";
+
+/** @todo - is there a way to have TS force us to make this always satisfy `KnowledgeGraphOutwardEdge`? */
+export type HasLinkEdge = {
```

Comment:
Should this be `HasOutgoingDirectedLinkEdge` so it's consistent with the type at the bottom? And potentially adding a `HasLeftEndpointEdge` so these are all consistent

Reply:
Hm, I'll change it for now because I think it will greatly increase usability. If we start introducing directions _on_ the link entities themselves though (for instance with bi-directional links) then these will become incorrect.

I think for now it's fine, as we don't have the thing I'm describing, just conscious of it.

> And potentially adding a HasLeftEndpointEdge so these are all consistent

Purposefully not including all variations at the moment, I think we should create them as we need and find useful, otherwise there'll be a big maintenance burden. Also I think we really want to figure out the `@todo` there before throwing loads of these in.

Reply:
[Rework edge aliases](https://github.com/hashintel/hash/pull/1438/commits/a6c56c7c29a42c1f169c34ccd6f297a505272e19)

Reply:
Thank you! it's looking nice. I'm fine with not including until we need them